src/Npgsql/NpgsqlConnector.cs

```diff
@@ -599,7 +599,12 @@ async Task RawOpen(NpgsqlTimeout timeout, bool async, CancellationToken cancella
         void Connect(NpgsqlTimeout timeout)
         {
             EndPoint[] endpoints;
-            if (!string.IsNullOrEmpty(Host) && Host[0] == '/')
+
+            // Unix-domain sockets are assumed when the host name:
+            // - starts with a forward slash (rooted paths on Unix, e. g. "/tmp")
+            // - has a colon as second character (rooted classical paths on Windows, e. g. "C:\my\socket\dir")
+            // - starts with a backslash (long paths or UNC paths on Windows, e. g. "\\?\D:\very long path" or "\\server\share")
+            if (!string.IsNullOrEmpty(Host) && (Host[0] == '/' || Host[1] == ':' || Host[0] == '\\'))
```

Comment:
Maybe mention that this is the libpq behavior

Reply:
> Maybe mention that this is the libpq behavior

It isn't.

For now libpq only looks for '/' which obviously makes it unusable on Windows.
Maybe I'll send a patch mimicking our behavior upstream.

Reply:
> It isn't.

Yes it is.

That's what they do:
```C
#define is_absolute_path(filename) \
( \
	IS_DIR_SEP((filename)[0]) || \
	(isalpha((unsigned char) ((filename)[0])) && (filename)[1] == ':' && \
	 IS_DIR_SEP((filename)[2])) \
)
#endif
```

At first I thought they are suffering from the same bug that I almost introduced but they aren't.
As they progress through the filename characters one by one they'd abort once they run into the terminating '\0'.

In fact they are even checking for the directory separator after the volume separator as per [this comment](https://github.com/postgres/postgres/blob/89a7d21dfc9d891d2b3dcfea161b58d4ea458af6/src/port/path.c#L414)
> On Win32, a drive letter _not_ followed by a slash, e.g. 'E:abc', is relative to the cwd on that drive, or the drive's root directory if that drive has no cwd.

I'm not sure if doing so is worthwhile on our side because it seems like a really rare corner case but we might decide to do this just for the sake of compatibility.

Reply:
> I'm not sure if doing so is worthwhile on our side because it seems like a really rare corner case but we might decide to do this just for the sake of compatibility.

Actually if it is true, they should probably even fix it in `System.Private.CoreLib`.